tests/unit/client.py

```diff
@@ -247,10 +247,41 @@ def avoid_network_race_condition():
     db_file = db_dir+'/haas.db'
     conn = sqlite3.connect(db_file)
     cur = conn.cursor()
+    timeout = 0
     while True:
         que = len(cur.execute('select * from networking_action;').fetchall())
         if que > 0:
             time.sleep(1)
+            timeout = timeout + 1
+            if timeout > 10:  # breakout if exceeds 10 secs.
+                break
+        else:
+            break
+
+
+def avoid_race_sqlalchemy():
+    from haas.model import Node, Project, NetworkingAction
+    from sqlalchemy import func
+    from sqlalchemy import create_engine
+    from sqlalchemy.orm import sessionmaker
+
+    global db_dir  # Value of db_dir is populated by `create_setup`.
+    uri = 'sqlite:///'+db_dir+'/haas.db'
+    engine = create_engine(uri)
+    Session = sessionmaker(bind=engine)
+    session = Session()
+    timeout = 0
+
+    while True:
```

Comment:
I will do the `for in range(10):` instead of a while loop. about the assert after the loop, I did that but then revised it to simply breaking out of the function when timeout exceeds, because that way the test fails anyways and with an appropriate message like 
`E           FailedAPICallException: A networking operation is already active on the nic.
`
Full output of such a failure test is given below. Let me know if you still want me to change the code to fail deliberately after the timeout period. ?

```
tests/unit/client.py F

==================================================== FAILURES =====================================================
____________________________________ Test_node.test_node_detach_network_error _____________________________________
self = <tests.unit.client.Test_node instance at 0x7f07914a1d88>

    def test_node_detach_network_error(self):
        C.node.connect_network('node-04', 'eth0', 'net-04', 'vlan/native')
    #    avoid_network_race_condition()
>       C.node.detach_network('node-04', 'eth0', 'net-04')

/home/sahil/git/HIL/tests/unit/client.py:385: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/home/sahil/git/HIL/haas/client/node.py:84: in detach_network
    self.httpClient.request('POST', url, data=payload)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <haas.client.node.Node object at 0x7f079140a490>, response = <Response [409]>

    def check_response(self, response):
        if response.ok:
            if response.request.method == 'GET':
                return response.json()
            else:  # For methods PUT, POST, DELETE
                return
        else:
            e = response.json()
>           raise FailedAPICallException(e['msg'])
E           FailedAPICallException: A networking operation is already active on the nic.

/home/sahil/git/HIL/haas/client/base.py:49: FailedAPICallException
============================================ 1 failed in 5.57 seconds =============================================
```